--- a/common.gypi
+++ b/common.gypi
@@ -59,8 +59,9 @@
             'cflags': [ '-O3', '-ffunction-sections', '-fdata-sections' ],
             'ldflags': [ '-Wl,--gc-sections' ],
           }, {
-            'cflags': [ '-O2', '-fno-strict-aliasing' ],
+            'cflags': [ '-Os', '-fno-strict-aliasing' ],
             'cflags!': [ '-O3', '-fstrict-aliasing' ],
+            'ldflags': [ '-Wl,--gc-sections' ],
             'conditions': [
               # Required by the dtrace post-processor. Unfortunately,
               # some gcc/binutils combos generate bad code when
@@ -68,7 +69,7 @@
               ['OS=="solaris"', {
                 'cflags': [ '-ffunction-sections', '-fdata-sections' ],
               }, {
-                'cflags!': [ '-ffunction-sections', '-fdata-sections' ],
+                'cflags': [ '-ffunction-sections', '-fdata-sections' ],
               }],
               ['clang == 0 and gcc_version >= 40', {
                 'cflags': [ '-fno-tree-vrp' ],
